{
  function.master=connection.getKeepAliveMaster();
  try {
    return function.call();
  }
 catch (  RemoteException re) {
    throw re.unwrapRemoteException();
  }
catch (  IOException e) {
    throw e;
  }
catch (  ServiceException se) {
    throw ProtobufUtil.getRemoteException(se);
  }
catch (  Exception e) {
    throw new IOException("Unexpected exception when calling master",e);
  }
 finally {
    function.master.close();
  }
}
